src/steamship/agents/examples/fact_learner.py

```diff
@@ -0,0 +1,75 @@
+import uuid
+from typing import List
+
+from steamship import Block
+from steamship.agents.llms.openai import OpenAI
+from steamship.agents.react import ReACTAgent
+from steamship.agents.schema import AgentContext
+from steamship.agents.schema.context import Metadata
+from steamship.agents.service.agent_service import AgentService
+from steamship.agents.tools.question_answering import VectorSearchQATool
+from steamship.agents.tools.question_answering.vector_search_learner_tool import (
+    VectorSearchLearnerTool,
+)
+from steamship.agents.utils import with_llm
+from steamship.invocable import post
+from steamship.utils.repl import AgentREPL
+
+
+class FactLearner(AgentService):
+    """FactLearner is an example AgentService contains an Agent which:
+
+    1) Learns facts to a vector store
+    2) Can answer questions based on those facts"""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._agent = ReACTAgent(
+            tools=[
+                VectorSearchLearnerTool(),
+                VectorSearchQATool(),
+            ],
+            llm=OpenAI(self.client),
+        )
+
+    @post("prompt")
+    def prompt(self, prompt: str) -> str:
```

Comment:
one minor comment here: when playing with history/memory, i quickly discovered the utility of being able to take "context id" as a param in the example method. we should be sure to add whatever mechanism we agree on in that PR to this bit.

Reply:
Yeah I really liked that element of those other PRs -- depending on the ordering of merge, I'll back-port it in!